File: pythonp/OOPS/py_composition/book.py
#Bibirectional relationship
class Book:
    def __init__(self, name):
        self.name = name


class Library:
    def __init__(self, name):
        self.name = name

    def addBooks(self, books):
        self.books = books

    def showBooks(self):
        for x in self.books:
            print(x.name)


b1 = Book("Blue")
b2 = Book("Red")
b3 = Book("Pink")

list = []
list.append(b1)
list.append(b2)
list.append(b3)

l1 = Library("Blue library")
l1.addBooks(list)
print(l1.showBooks())

File: javap/OOPS/j_composition/book.java
package javap.OOPS.j_composition;

import java.util.List;
import java.util.ArrayList;

//Bidirectional relationship.
class Book {
    String name;

    Book(String name) {
        this.name = name;
    }
}

class Library {
    String name;
    List<Book> books;

    Library(String name) {
        this.name = name;
    }

    String getLibrary() {
        return this.name;
    }

    void setBooks(List<Book> books) {
        this.books = books;
    }

    List<Book> getBooks() {
        return this.books;
    }
}

class Operations {
    public static void main(String args[]) {
        Library l1 = new Library("Bell Library");
        Book b1 = new Book("New 1");
        Book b2 = new Book("New 2");
        Book b3 = new Book("New 3");
        List<Book> books = new ArrayList<Book>();
        books.add(b1);
        books.add(b2);
        books.add(b3);
        l1.setBooks(books);
        System.out.println("Library :: " + l1.getLibrary());
        System.out.println("Books :: " + l1.getBooks().get(1).name);
    }
}